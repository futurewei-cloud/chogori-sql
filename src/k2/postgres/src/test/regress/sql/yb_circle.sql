--
-- CIRCLE
--

CREATE TABLE CIRCLE_TBL (id int primary key, f1 circle);

INSERT INTO CIRCLE_TBL VALUES (1, '<(5,1),3>');

INSERT INTO CIRCLE_TBL VALUES (2, '<(1,2),100>');

INSERT INTO CIRCLE_TBL VALUES (3, '1,3,5');

INSERT INTO CIRCLE_TBL VALUES (4, '((1,2),3)');

INSERT INTO CIRCLE_TBL VALUES (5, '<(100,200),10>');

INSERT INTO CIRCLE_TBL VALUES (6, '<(100,1),115>');

-- bad values

INSERT INTO CIRCLE_TBL VALUES (7, '<(-100,0),-100>');

INSERT INTO CIRCLE_TBL VALUES (8, '1abc,3,5');

INSERT INTO CIRCLE_TBL VALUES (9, '(3,(1,2),3)');

SELECT * FROM CIRCLE_TBL ORDER BY id;

SELECT '' AS six, center(f1) AS center
  FROM CIRCLE_TBL ORDER BY id;

SELECT '' AS six, radius(f1) AS radius
  FROM CIRCLE_TBL ORDER BY id;

SELECT '' AS six, diameter(f1) AS diameter
  FROM CIRCLE_TBL ORDER BY id;

SELECT '' AS two, f1 FROM CIRCLE_TBL WHERE radius(f1) < 5 ORDER BY id;

SELECT '' AS four, f1 FROM CIRCLE_TBL WHERE diameter(f1) >= 10 ORDER BY id;

SELECT '' as five, c1.f1 AS one, c2.f1 AS two, (c1.f1 <-> c2.f1) AS distance
  FROM CIRCLE_TBL c1, CIRCLE_TBL c2
  WHERE (c1.f1 < c2.f1) AND ((c1.f1 <-> c2.f1) > 0)
  ORDER BY distance, area(c1.f1), area(c2.f1);
