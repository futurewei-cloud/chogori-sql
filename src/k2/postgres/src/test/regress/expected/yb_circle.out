--
-- CIRCLE
--
CREATE TABLE CIRCLE_TBL (id int primary key, f1 circle);
INSERT INTO CIRCLE_TBL VALUES (1, '<(5,1),3>');
INSERT INTO CIRCLE_TBL VALUES (2, '<(1,2),100>');
INSERT INTO CIRCLE_TBL VALUES (3, '1,3,5');
INSERT INTO CIRCLE_TBL VALUES (4, '((1,2),3)');
INSERT INTO CIRCLE_TBL VALUES (5, '<(100,200),10>');
INSERT INTO CIRCLE_TBL VALUES (6, '<(100,1),115>');
-- bad values
INSERT INTO CIRCLE_TBL VALUES (7, '<(-100,0),-100>');
ERROR:  invalid input syntax for type circle: "<(-100,0),-100>"
LINE 1: INSERT INTO CIRCLE_TBL VALUES (7, '<(-100,0),-100>');
                                          ^
INSERT INTO CIRCLE_TBL VALUES (8, '1abc,3,5');
ERROR:  invalid input syntax for type circle: "1abc,3,5"
LINE 1: INSERT INTO CIRCLE_TBL VALUES (8, '1abc,3,5');
                                          ^
INSERT INTO CIRCLE_TBL VALUES (9, '(3,(1,2),3)');
ERROR:  invalid input syntax for type circle: "(3,(1,2),3)"
LINE 1: INSERT INTO CIRCLE_TBL VALUES (9, '(3,(1,2),3)');
                                          ^
SELECT * FROM CIRCLE_TBL ORDER BY id;
 id |       f1
----+----------------
  1 | <(5,1),3>
  2 | <(1,2),100>
  3 | <(1,3),5>
  4 | <(1,2),3>
  5 | <(100,200),10>
  6 | <(100,1),115>
(6 rows)

SELECT '' AS six, center(f1) AS center
  FROM CIRCLE_TBL ORDER BY id;
 six |  center
-----+-----------
     | (5,1)
     | (1,2)
     | (1,3)
     | (1,2)
     | (100,200)
     | (100,1)
(6 rows)

SELECT '' AS six, radius(f1) AS radius
  FROM CIRCLE_TBL ORDER BY id;
 six | radius
-----+--------
     |      3
     |    100
     |      5
     |      3
     |     10
     |    115
(6 rows)

SELECT '' AS six, diameter(f1) AS diameter
  FROM CIRCLE_TBL ORDER BY id;
 six | diameter
-----+----------
     |        6
     |      200
     |       10
     |        6
     |       20
     |      230
(6 rows)

SELECT '' AS two, f1 FROM CIRCLE_TBL WHERE radius(f1) < 5 ORDER BY id;
 two |    f1
-----+-----------
     | <(5,1),3>
     | <(1,2),3>
(2 rows)

SELECT '' AS four, f1 FROM CIRCLE_TBL WHERE diameter(f1) >= 10 ORDER BY id;
 four |       f1
------+----------------
      | <(1,2),100>
      | <(1,3),5>
      | <(100,200),10>
      | <(100,1),115>
(4 rows)

SELECT '' as five, c1.f1 AS one, c2.f1 AS two, (c1.f1 <-> c2.f1) AS distance
  FROM CIRCLE_TBL c1, CIRCLE_TBL c2
  WHERE (c1.f1 < c2.f1) AND ((c1.f1 <-> c2.f1) > 0)
  ORDER BY distance, area(c1.f1), area(c2.f1);
 five |      one       |      two       |     distance
------+----------------+----------------+------------------
      | <(100,200),10> | <(100,1),115>  |               74
      | <(100,200),10> | <(1,2),100>    | 111.370729772479
      | <(1,3),5>      | <(100,200),10> | 205.476756144497
      | <(5,1),3>      | <(100,200),10> |  207.51303816328
      | <(1,2),3>      | <(100,200),10> | 208.370729772479
(5 rows)
